NMC-533 Set new folder name inline: rebased

skjnldsv opened the pull request:
Rebased version of https://github.com/nextcloud/server/pull/28645

Closes https://github.com/nextcloud/server/pull/28645

TSI-kavitasonawane committed:
Implementation of Add button folder creation and Add button text document creation

Signed-off-by: Kavita Sonawane <kavita.sonawane@t-systems.com>

TSI-kavitasonawane committed:
Prevents the sidebar from being shown when creating a new directory

Signed-off-by: Kavita Sonawane <kavita.sonawane@t-systems.com>

skjnldsv committed:
Fix template picker styling with new Modal size

Signed-off-by: John Molakvoæ <skjnldsv@protonmail.com>

szaimen commented:
@skjnldsv are the styling changes to the template picker really necessary? How does it look like on mobile now?

szaimen commented on apps/files/src/views/TemplatePicker.vue:
```diff
@@ -284,9 +280,12 @@ export default {
+	@media only screen and (min-width: calc(1024px / 2 + 1px)) {
+		::v-deep .modal-container {
+			position: relative;
+			height: fit-content !important;
```
Okay, fine by me then

szaimen commented on apps/files/src/views/TemplatePicker.vue:
```diff
@@ -24,7 +24,7 @@
 		:clear-view-delay="-1"
 		class="templates-picker"
-		size="large"
+		size="normal"
```
Jup

szaimen reviewed (approved):
tested and seems to work well 👍
(didn't review the code though)

skjnldsv commented:
/rebase

skjnldsv commented:
@danxuliu can yuo help with the acceptance tests please?

szaimen commented on apps/files/src/views/TemplatePicker.vue:
```diff
@@ -284,9 +280,12 @@ export default {
+	@media only screen and (min-width: calc(1024px / 2 + 1px)) {
+		::v-deep .modal-container {
+			position: relative;
+			height: fit-content !important;
+		}
```
I think we should be able to reverts this change after https://github.com/nextcloud/server/pull/31916
So rebasing and not changing anything here should work, IIRC.

szaimen replied:
done

danxuliu commented:
> @danxuliu can yuo help with the acceptance tests please?

I would love to, but unless I find a time-turner somewhere it is unlikely that I will be able to do it this week, sorry.

danxuliu committed:
Adjust acceptance tests to changes in new folder creation

Before, to create a new folder, the name of the folder was set in the
menu item and then the folder was created. Now the new folder is created
with a default name and, once it appears in the file list, its name is
in editable mode to set the desired one.

The default folder name is not "New folder" but "New folder (X)" if
there was already a folder named "New folder", so "New folder" can not
be used to look for the new entry. However, there will be only one file
list entry with an editable name at each time, so that is used instead
to look for the new entry after the new folder is created.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

nextcloud-command committed:
Compile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

danxuliu committed:
Adjust acceptance tests to details view state after creating folder

The details view is no longer automatically opened when a folder is
created.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu commented:
> > @danxuliu can yuo help with the acceptance tests please?
> 
> I would love to, but unless I find a time-turner somewhere it is unlikely that I will be able to do it this week, sorry.

Acceptance tests should be ready now.

szaimen commented:
/compile amend /

szaimen commented:
@skjnldsv could you please have a look if all changes are still present? I initially messed up the rebase...

danxuliu commented:
> Looks like one scenario is still failing: ![image](https://user-images.githubusercontent.com/42591237/163968635-43a010b8-43fd-4899-ba0b-9b8bd04d5cff.png)

Although that scenario is failing that one is not the real failure ;-) In errors like those the acceptance tests try the action again, as you can see from the following line to the ones in the screenshot (`Trying again`). As there is not another error immediately after trying again the action succeeded and the test continued.

The real failure is in the last step:
```
And I see that the file list contains a file named "Subsubfolder" # FileListContext::iSeeThatTheFileListContainsAFileNamed()
      Row for file Subsubfolder in file list could not be found after 100 seconds (NoSuchElementException)
```

The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(

Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.

szaimen commented:
> Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.

Sorry for that. Shall I do this here or is it fine for now?

szaimen commented:
> The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(

do we wanna try to restart drone or hope that it was only a hiccup?

danxuliu commented:
> > Independently of all that, if the last commit does not touch JavaScript files please do not amend it to add the compiled files and add an independent commit instead.
> 
> Sorry for that. Shall I do this now or is it fine for now?

No problem at all :-) I have split the commit in two, it is fine :+1: 

> > The test passes locally, so it looks like a hiccup. I would like to understand why it failed to harden the test, but unfortunately I can not see how it could have happened :-(
>
> do we wanna try to restart drone or hope that it was only a hiccup?

As I have force pushed to split the commit (twice, because the first time I used the wrong commit author :facepalm:) Drone will run again (I have already stopped the first run corresponding to the commit with the wrong author). So let's see ;-)

szaimen commented:
Great, thanks! :)

szaimen commented:
It seems to fail again...

danxuliu commented:
> It seems to fail again...

It is just another hiccup :shrug: And it did not fail in the same step, but on a previous one (looking for _Subfolder_ rather than for _Subsubfolder_). Hopefully I will be able to find out what is going on at some point...

In any case, from the point of view of the acceptance tests, the pull request could be merged despite the failure.

szaimen commented:
> In any case, from the point of view of the acceptance tests, the pull request could be merged despite the failure.

Okay 👍

I'll leave that to @skjnldsv  :)

szaimen commented:
is this readdy to merge?
